apps/halide/blur/main.cpp

```diff
@@ -96,40 +96,23 @@ bool write_png_file(
 }
 
 typedef void (*blurtype)(
-    void *ctxt, int_fast32_t n, int_fast32_t m, uint8_t *g, const uint8_t *inp);
+    void *ctxt, int_fast32_t n, int_fast32_t m, uint16_t *g, const uint16_t *inp);
 
 int exec_parrot(blurtype func, std::string output_name, int width, int height,
-    uint8_t *parrot) {
-  uint8_t *parrot_blurred_0, *parrot_blurred_1;
-  parrot_blurred_0 =
-      (uint8_t *)malloc(sizeof(uint8_t) * (width + 4) * (height + 4));
-  parrot_blurred_1 =
-      (uint8_t *)malloc(sizeof(uint8_t) * (width + 4) * (height + 4));
+    uint16_t *parrot) {
+  uint16_t *parrot_write = (uint16_t *)malloc(sizeof(uint16_t) * width * height);
 
   auto start = std::chrono::steady_clock::now();
   int iterations = 100;
   for (int i = 0; i < iterations; i += 4) {
```

Comment:
Why are you skipping 4 iterations at a time? It seems like this will mean you are doing an incorrect average later on because you are assuming you are running the program `iterations` times.

Reply:
oops accident from Yuka's code